contract/src/bitcoin.rs

```diff
@@ -0,0 +1,92 @@
+use crate::utils::BlocksGetter;
+use crate::{BtcLightClient, BtcLightClientExt, Header, U256};
+use btc_types::header::ExtendedHeader;
+use btc_types::utils::target_from_bits;
+use near_sdk::{near, require};
+
+#[near]
+impl BtcLightClient {
+    pub(crate) fn check_pow(&self, block_header: &Header, prev_block_header: &ExtendedHeader) {
+        let expected_bits = self.get_next_work_required(block_header, prev_block_header);
+
+        require!(
+            expected_bits == block_header.bits,
+            format!(
+                "Error: Incorrect target. Expected bits: {:?}, Actual bits: {:?}",
+                expected_bits, block_header.bits
+            )
+        );
+    }
+
+    //https://github.com/bitcoin/bitcoin/blob/ae024137bda9fe189f4e7ccf26dbaffd44cbbeb6/src/pow.cpp#L14
+    fn get_next_work_required(
+        &self,
+        block_header: &Header,
+        prev_block_header: &ExtendedHeader,
```

Comment:
Let's move out `get_next_work_required` and `calculate_next_work_required` from the `BtcLightClient` and pass them as a trait. This will simplify the migration of tests from the original source.

Reply:
fixed: https://github.com/Near-One/btc-light-client-contract/pull/101/commits/028c9123b124b264a6a8359b6daa629e13aade24